include/LightGBM/config.h

```diff
@@ -473,15 +473,34 @@ struct ParameterAlias {
     });
     std::unordered_map<std::string, std::string> tmp_map;
     for (const auto& pair : *params) {
-      if (alias_table.count(pair.first) > 0) {
-        tmp_map[alias_table.at(pair.first)] = pair.second;
-      } else if (parameter_set.count(pair.first) == 0) {
-        Log::Fatal("Unknown parameter: %s", pair.first.c_str());
+      auto alias = alias_table.find(pair.first);
+      if (alias != alias_table.end()) { // found alias
+        auto alias_set = tmp_map.find(alias->second); 
+        if (alias_set != tmp_map.end()) { // alias already set
+          // set priority alphabetically to ensure reproducible behavior
```

Comment:
maybe using the strlen for the priority  is better? since users often use the short names.